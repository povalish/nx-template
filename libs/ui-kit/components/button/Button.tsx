/* eslint-disable react/button-has-type */
import React from 'react';
import { IWebButton } from './Button.interface';
import { textClasses, buttonClasses } from './Button.classes';
import { cx } from 'class-variance-authority';

//
//

export const Button: React.FC<IWebButton> = ({ children, className, ...buttonProps }) => (
  <button className={cx(textClasses(), buttonClasses({ className }))} {...buttonProps}>
    {children}
  </button>
);
