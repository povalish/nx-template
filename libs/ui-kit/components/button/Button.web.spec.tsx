import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './Button';

//
//

describe('Button component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Button>Click me!</Button>);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me!</Button>);

    fireEvent.click(screen.getByText(/click me/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
