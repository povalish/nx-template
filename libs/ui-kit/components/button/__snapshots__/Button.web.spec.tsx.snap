// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render successfully 1`] = `
<body>
  <div>
    <button
      class="text-center font-bold text-colorTextContrast 
  py-2 px-3 min-w-[100px]
  border-[3px] rounded-lg border-red 
  transition-all
  active:scale-95
  hover:scale-105
  disabled:bg-colorDisabled disabled:border-colorDisabled disabled:scale-100 disabled:cursor-not-allowed
  focus:border-[3px] focus:border-colorAccent focus:scale-105 focus:shadow-focus focus:outline-none
 
          border-colorSuccess bg-colorSuccess 
          hover:bg-colorSuccessHover 
          active:bg-colorSuccessPressed
        "
    >
      Click me!
    </button>
  </div>
</body>
`;
